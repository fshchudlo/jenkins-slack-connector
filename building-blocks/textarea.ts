import { textbox } from "./textbox";

export function textarea(label: string, action_id: string, initial_value: string | null = null, placeholder: string | null = null) {
    return textbox(label, action_id, initial_value, placeholder, true);
}