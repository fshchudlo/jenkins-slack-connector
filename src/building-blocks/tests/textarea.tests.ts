import { textarea } from "../textarea";

describe("textarea", () => {
    it("should create a textarea with the correct properties", () => {
        const control = textarea("Enter text", "action_id", "initial value", "placeholder");
        expect(control).toMatchSnapshot();
    });

    it("should handle empty initial_value and placeholder", () => {
        const control = textarea("Enter text", "action_id");
        expect(control.element.initial_value).toBeFalsy();
        expect(control.element.placeholder).toBeUndefined();
    });
});
