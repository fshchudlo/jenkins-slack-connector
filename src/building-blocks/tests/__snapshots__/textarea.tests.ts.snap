// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`textarea should create a textarea with the correct properties 1`] = `
{
  "block_id": "action_id",
  "element": {
    "action_id": "action_id",
    "initial_value": "initial value",
    "multiline": true,
    "placeholder": {
      "text": "placeholder",
      "type": "plain_text",
    },
    "type": "plain_text_input",
  },
  "label": {
    "text": "Enter text",
    "type": "plain_text",
  },
  "type": "input",
}
`;
